pcdsdevices/tmo_ip1.py

```diff
@@ -0,0 +1,50 @@
+"TMO sample calibration foil to work with spectrometer"
+from lightpath import LightpathState
+from ophyd.device import Component as Cpt
+
+from .device import GroupDevice
+from .device import UpdateComponent as UpCpt
+from .epics_motor import BeckhoffAxis
+from .interface import BaseInterface, LightpathMixin
+from .pmps import TwinCATStatePMPS
+
+
+class CalibrationAxis(TwinCATStatePMPS):
+    """
+    Sample calibration 1D State Setup
+    Here, we specify 5 states, and 1 motor, for Y
+    axe.
+    """
+    config = UpCpt(state_count=5, motor_count=1)
+
+
+class SCaliFoil(BaseInterface, GroupDevice, LightpathMixin):
+    """
+    TMO PMPS1D sample calibration class.
+    Parameters:
+    -----------
+    prefix : str
+        Base PV for the motion system
+    name : str
+        Alias for the device
+    """
+    # UI Representation
+    _icon = 'fa.minus-square'
+    tab_component_names = True
+
+    # Sample calibration Y axe
+    cali_foil = Cpt(CalibrationAxis, 'AT2K4:IP1:STATE', add_prefix=(), kind='normal')
+    cali_foil_y = Cpt(BeckhoffAxis, ':MMS', doc="Y-axis of sample caliration at2k4", kind='normal')
```

Comment:
My old comment about this is auto-marked as outdated now, but it's still relevant.

I don't like how one of these is hard-coded and the other is not.
In this case, I would suggest that they should both be hard-coded.

If only one is hard-coded, it will be possible to pick a wrong prefix and get a half-working device.